analyzers/src/SonarAnalyzer.CSharp/Rules/UnnecessaryUsings.cs

```diff
@@ -85,6 +85,12 @@ private static void CheckUnnecessaryUsings(SyntaxNodeAnalysisContext context, IE
         {
             foreach (var usingDirective in usingDirectives)
             {
+                // This will create some FNs but will kill noise from FPs.
+                // For more info see issue: https://github.com/SonarSource/sonar-dotnet/issues/5946
+                if (usingDirective.GetFirstToken().IsKind(SyntaxKind.GlobalKeyword))
```

Comment:
I suggest to merge the two ifs to increase the readability. If you agree with it do it otherwise it is good to go.